frontend/src/app/auth.service.ts

```diff
@@ -19,26 +19,24 @@ export class AuthService {
   public user$: Observable<User | null>;
 
   constructor(
-    private http: Http,
+    private http: HttpClient,
     private router: Router,
   ) {
     this.changeUserEvent = new BehaviorSubject(null);
     this.user$ = this.changeUserEvent.asObservable();
 
-    this.http.get(this._currentUserUrl, { withCredentials: true })
-      .subscribe(rsp => {
-        this._currentUser = (rsp.text() === '' ? null : rsp.json());
+    this.http.get<User>(this._currentUserUrl)
```

Comment:
As I point in Django part of this PR it not always return User (it can be logged out). Do You want to return `404` there just to not go to `subscribe` here?